### ISSUE TITLE
Unreal Engine 5 Probes tab "Python is required for Project Acoustics baking. Please enable the Python plugin."

### ISSUE THREAD

**Issue by mpierluissi** [microsoft/ProjectAcoustics#120]
Currently facing an issue with Project Acoustics and Python plugins not communicating. Whenever I enter the Probes tab, I only see "Python is required for Project Acoustics baking. Please enable the Python plugin."

Currently enabled in UE5:
- Python Editor Script Plugin
- Sequencer Scripting

Installed in Visual Studio Installer (Visual Studio Community 2019):
- Visual Studio core editor
- Python development
- .NET desktop development
- Desktop development with C++
- Game development with Unity
- Game development with C++
- .NET cross-platform development

System:
- Windows 10 Education 21H2
- AMD Ryzen 9 5900X

Clicking on the Bake tab throws this error and crashes UE5:

```
LoginId:89e19567422e479ca9bf33928009deff
EpicAccountId:11e776cb37c048d9aab3ef18d91d6147

Assertion failed: IsValid() [File:D:\RocketSync\5.0.0-19505902+++UE5+Release-5.0\Working\Engine\Source\Runtime\Core\Public\Templates\SharedPointer.h] [Line: 1074]

UnrealEditor_ProjectAcousticsBakeUI!SAcousticsBakeTab::Refresh() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\ProjectAcoustics\Source\ProjectAcousticsBakeUI\Private\AcousticsBakeTab.cpp:217]
UnrealEditor_ProjectAcousticsBakeUI!FAcousticsEdMode::OnClickBakeTab() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\ProjectAcoustics\Source\ProjectAcousticsBakeUI\Public\AcousticsEdMode.h:111]
UnrealEditor_ProjectAcousticsBakeUI!SAcousticsEdit::OnActiveTabChanged() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\ProjectAcoustics\Source\ProjectAcousticsBakeUI\Public\SAcousticsEdit.h:90]
UnrealEditor_ProjectAcousticsBakeUI!TBaseSPMethodDelegateInstance<0,SAcousticsEdit,1,void __cdecl(enum AcousticsActiveTab),FDefaultDelegateUserPolicy>::ExecuteIfSafe() [D:\RocketSync\5.0.0-19505902+++UE5+Release-5.0\Working\Engine\Source\Runtime\Core\Public\Delegates\DelegateInstancesImpl.h:309]
UnrealEditor_ProjectAcousticsBakeUI!SSegmentedControl<enum AcousticsActiveTab>::CommitValue() [D:\RocketSync\5.0.0-19505902+++UE5+Release-5.0\Working\Engine\Source\Runtime\Slate\Public\Widgets\Input\SSegmentedControl.h:540]
UnrealEditor_ProjectAcousticsBakeUI!UE::Core::Private::Tuple::TTupleBase<TIntegerSequence<unsigned int,0>,enum AcousticsActiveTab>::ApplyAfter<void (__cdecl SSegmentedControl<enum AcousticsActiveTab>::*const &)(enum ECheckBoxState,enum AcousticsActiveTab),SS() [D:\RocketSync\5.0.0-19505902+++UE5+Release-5.0\Working\Engine\Source\Runtime\Core\Public\Templates\Tuple.h:324]
UnrealEditor_ProjectAcousticsBakeUI!TBaseSPMethodDelegateInstance<0,SSegmentedControl<enum AcousticsActiveTab>,1,void __cdecl(enum ECheckBoxState),FDefaultDelegateUserPolicy,enum AcousticsActiveTab>::ExecuteIfSafe() [D:\RocketSync\5.0.0-19505902+++UE5+Release-5.0\Working\Engine\Source\Runtime\Core\Public\Delegates\DelegateInstancesImpl.h:309]
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_Slate
UnrealEditor_ApplicationCore
UnrealEditor_ApplicationCore
UnrealEditor_ApplicationCore
UnrealEditor_ApplicationCore
user32
user32
InkObj
atlthunk
user32
user32
UnrealEditor_ApplicationCore
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
kernel32
ntdll
```

Attempted solution #89 and from the [Unreal Forums](https://forums.unrealengine.com/t/unreal-project-acoustics-python-problem/562079/2) to no avail.

Thank you for your time!

**Comment by KyleStorck** [comment]
Did you try the hacky workaround from this comment? https://github.com/microsoft/ProjectAcoustics/issues/89#issuecomment-1132307029

**Comment by KyleStorck** [comment]
And just to confirm, these specifically are the 2 components that you need along with VS 2019.

- .NET Framework 4.0 targeting pack
- .NET desktop environment

**Comment by mpierluissi** [comment]
Thank you for the quick reply! Yes I attempted #89 with both --upgrade pythonnet and installing the .whl file. The install results in this error:
![Pythonnet_fail](https://user-images.githubusercontent.com/66741934/193933851-41b576dc-a533-40d9-9906-cae5a7eef3c2.PNG)
Perhaps I am not installing it correctly?

As for the VS 2019 installs, I have the .NET Framework 4.0 targeting pack, but I am unable to find the desktop _environment_ specifically.

**Comment by KyleStorck** [comment]
Might have a solution for the install error. Described in the second part of this comment. https://github.com/microsoft/ProjectAcoustics/issues/102#issuecomment-1194078117 You might need to purge to guarantee that you are allowed to install fresh

**Comment by mpierluissi** [comment]
SOLVED: Did not run Command Prompt as administrator.

After running as admin, I performed the #89 solution again and installed the .whl properly. Now the Probes tab is working!
![Pythonnet_success](https://user-images.githubusercontent.com/66741934/193935962-bdfde319-78eb-49dd-a47a-3b4ceb8bd664.PNG)

Thank you @KyleStorck for helping me try it again and find the obvious!

**Comment by KyleStorck** [comment]
Glad to hear. That's another solution we need to remember

**Comment by MikeChemi** [comment]
While the workaround used in this thread works (and if you are in a working state please do not change anything!), I wanted to give more information into what seems to have happened.

Given the timing of this issue I suspect this problem was related to the new version of Pythonnet. Pythonnet released version 3.0 12 days ago, which has support for newer python versions. Unfortunately, this doesn't work out of the box with the Project Acoustics plugin as there was an API change in Pythonnet. You can update the python script included in the Project Acoustics plugin at Plugins\ProjectAcousticsNative\Content\Python\init_project_acoustics.py to add the following two lines:

clr.AddReference("System.Security")
clr.AddReference("Microsoft.Azure.Batch")

You'll want to add these lines after the line:
from System import AppDomain, Convert, String

Make the python script changes before launching the editor, and you should be good to go with the latest pythonnet.
